coderd/workspaceapps/proxy.go

```diff
@@ -395,41 +397,73 @@ func (s *Server) HandleSubdomain(middlewares ...func(http.Handler) http.Handler)
 				return
 			}
 
-			// Use the passed in app middlewares before checking authentication and
```

Comment:
Note to reviewers: this is the meat of the change.
I had to move some things around here since the token was only being created _after_ CORS was handled, and the defined behavior is stored in the token.

I moved the CORS middleware invocation into `determineCORSBehavior` because we needed to conditionally invoke it if behavior != `passthru`, only.